File: merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ans;
        if(intervals.size() == 0){
            return ans;
        }
        sort(intervals.begin(), intervals.end());
        vector<int> temp = intervals[0];
        for(auto x: intervals){
            if(x[0] <= temp[1]){
                temp[1] = max(x[1],temp[1]);
            }else{
                ans.push_back(temp);
                temp = x;
            }
        }
        ans.push_back(temp);
        return ans;
    }
};

File: merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparing(a -> a[0]));
        List<int[]> arr = new ArrayList<>();
        for(int[] interval : intervals){
            if(arr.isEmpty() || arr.get(arr.size() - 1)[1] < interval[0]){
                arr.add(interval);
            }else{
                arr.get(arr.size() - 1)[1] = Math.max(arr.get(arr.size() - 1)[1], interval[1]);
            }
        }
        return arr.toArray(new int[arr.size()][]);
    }
}